src/snowflake/snowpark/mock/_telemetry.py

```diff
@@ -0,0 +1,206 @@
+TELEMETRY_KEY_PYTHON_VERSION = "Python_Version"
+TELEMETRY_KEY_EVENT_TYPE = "Event_type"
+TELEMETRY_KEY_CONN_UUID = "Connection_UUID"
+TELEMETRY_KEY_FEATURE_NAME = "feature_name"
+TELEMETRY_KEY_PARAMETERS_INFO = "parameters_info"
+TELEMETRY_KEY_ERROR_MESSAGE = "error_message"
+TELEMETRY_KEY_IS_INTERNAL = "is_internal"
+
+
+def generate_base_oob_telemetry_data_dict(
+    connection_uuid: Optional[str] = None,
+) -> dict:
+    return {
+        TELEMETRY_KEY_TYPE: TELEMETRY_VALUE_SNOWPARK_EVENT_TYPE,
+        TELEMETRY_KEY_UUID: str(uuid.uuid4()),
+        TELEMETRY_KEY_CREATED_ON: str(datetime.now().strftime("%Y-%m-%d %H:%M:%S")),
+        TELEMETRY_KEY_MESSAGE: {TELEMETRY_KEY_CONN_UUID: connection_uuid},
+        TELEMETRY_KEY_TAGS: {
+            TELEMETRY_KEY_SNOWPARK_VERSION: SNOWPARK_PYTHON_VERSION,
+            TELEMETRY_KEY_OS_VERSION: OS_VERSION,
+            TELEMETRY_KEY_PYTHON_VERSION: PYTHON_VERSION,
+        },
+        TELEMETRY_KEY_PROPERTIES: {
+            TELEMETRY_KEY_LOWER_TYPE: TELEMETRY_VALUE_SNOWPARK_EVENT_TYPE
+        },
+    }
+
+
+class LocalTestTelemetryEventType(Enum):
+    UNSUPPORTED = "unsupported"
+    SUPPORTED = "supported"
+    SESSION_CONNECTION = "session"
+
+
+class LocalTestOOBTelemetryService(TelemetryService):
+    PROD = "https://client-telemetry.c1.us-west-2.aws.app.snowflake.com/enqueue"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._is_internal_usage = bool(
+            os.getenv("SNOWPARK_LOCAL_TESTING_INTERNAL_TELEMETRY", False)
+        )
+        self._deployment_url = self.PROD
+
+    def _upload_payload(self, payload) -> None:
+        success = True
+        response = None
+        try:
+            with requests.Session() as session:
+                response = session.post(
+                    self._deployment_url,
+                    data=payload,
+                    timeout=REQUEST_TIMEOUT,
+                    headers={"Content-type": "application/json"},
+                )
+                if (
+                    response.status_code == OK
+                    and json.loads(response.text).get("statusCode", 0) == OK
+                ):
+                    logger.debug(
+                        "telemetry server request success: %d", response.status_code
+                    )
+                else:
+                    logger.debug(
+                        "telemetry server request error: %d", response.status_code
+                    )
+                    success = False
+        except Exception as e:
+            logger.debug(
+                "Telemetry request failed, Exception response: %s, exception: %s",
+                response,
+                str(e),
+            )
+            success = False
+        finally:
+            logger.debug("Telemetry request success=%s", success)
+
+    def add(self, event) -> None:
+        """Adds a telemetry event to the queue. If the event is urgent, upload all telemetry events right away."""
```

Comment:
I don't see anything that would do an immediate upload.

Reply:
sorry about the comment being confusing -- I copied it from the connector telemetry code.
I removed the comment.

I don't think the concept "urgent" is applicable to local test, doing batch sending should be enough for local testing telemetry events. in the connector, urgent is for ocsp validation error.